src/app/_components/TaskList.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from "react";
+
+interface checkListItem {
+  id: number;
+  task: string;
+  isChecked: boolean;
+}
+
+export default function TaskList() {
+  const [taskList, setTaskList] = useState<checkListItem[]>([]);
+  const [task, setNewTask] = useState<string>("");
+  const [incrementID, setIncrementID] = useState<number>(0);
+
+  const addedTasks = (text: string) => {
+    const newTaskList: checkListItem = {
+      id: incrementID,
+      task: text,
+      isChecked: false,
+    };
+
+    setTaskList([...taskList, newTaskList]);
+    setNewTask("");
+    setIncrementID(incrementID + 1);
+  };
```

Comment:
Instead of incrementing the ID based on state, you can still run into the issue of creating duplicates when the component remounts and the state is reset back to 0. I guess you can use react-uuid to create an id for the task as well as assign that same ID as a key to the `<li>`

Reply:
got it!